kube-core/src/dynamic.rs

```diff
@@ -1,14 +1,28 @@
 //! Contains types for using resource kinds not known at compile-time.
 //!
 //! For concrete usage see [examples prefixed with dynamic_](https://github.com/kube-rs/kube/tree/main/examples).
-
 pub use crate::discovery::ApiResource;
 use crate::{
     metadata::TypeMeta,
     resource::{DynamicResourceScope, Resource},
 };
+
 use k8s_openapi::apimachinery::pkg::apis::meta::v1::ObjectMeta;
 use std::borrow::Cow;
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+#[error("failed to parse this DynamicObject into a Resource")]
```

Comment:
Should probably include info on the error Display as we do elsewhere:

```sh
$ rg serde_json::Error -B 1
kube-client/src/client/auth/oauth.rs
43-    #[error("failed to parse token: {0}")]
44:    ParseToken(#[source] serde_json::Error),
```

Reply:
I think I have addressed this now, by including the source in the error message, let me know if this is not what you were after.
```rust
#[error("failed to parse this DynamicObject into a Resource: {source}")]
/// Failed to parse `DynamicObject` into `Resource`
pub struct ParseDynamicObjectError {
    #[from]
    source: serde_json::Error,
```

Reply:
Yeah, that looks good :-)